2-spending-db/task.md

```diff
@@ -13,37 +13,54 @@ To submit this homework write the correct commands after each question.
 
 ### 1. Show the date, transaction_no, description and amount for those transactions whose amount is between £30,000 and £31,000.
 ```sql
-
+SELECT date, transaction_no, description, amount 
+FROM spends 
+WHERE amount BETWEEN 30000 AND 31000;
 ```
 
 ### 2. Show the date, transaction_no, supplier_inv_no, description and amount for those transactions whose description includes the word 'fee'.
 ```sql
-
+select date, transaction_no, description, amount 
+from spends
+where description LIKE '%fee%';
 ```
 
 ### 3. Show the date, transaction_no, supplier_inv_no, description and amount for those transactions whose description includes the word 'Fee'.
 ```sql
-
+select date, transaction_no, description, amount 
+from spends
+where description LIKE '%Fee%';
 ```
 
 ### 4. Show the date, transaction_no, supplier_inv_no, description and amount for those transactions whose description includes the word 'fee' (case insensitive). You will need to search 'https://www.postgresql.org/docs/' to solve this.
 ```sql
-
+select date, transaction_no, description, amount 
+from spends
+where lower(description) LIKE '%fee%';
 ```
 
 ### 5. Show the date, transaction_no, supplier_inv_no, description and amount for those transactions whose amount is £25,000, £30,000, £35,000 or £40,000.
 ```sql
-
+select date, transaction_no, description, amount 
+from spends
+where amount IN (25000, 30000, 35000, 40000);
 ```
 
 ### 6. Show the date, the supplier_id, the description and the amount for transactions with the expense area of 'Better Hospital Food'. You could do a query to get the expense_area_id first then do a query to find the dates, supplier_ids and amounts. But it would be better to do this all in one query by linking the tables together using INNER JOINs.
 ```sql
-
+SELECT s.date, s.supplier_id, s.description, s.amount
+FROM spends s
+INNER JOIN expense_areas ea ON s.expense_area_id = ea.id
+WHERE ea.expense_area = 'Better Hospital Food';
 ```
 
 ### 7. Show the date, supplier name, description and amount for transactions with the expense area of 'Better Hospital Food'. You will need to INNER JOIN another table to be able to do this.
 ```sql
-
+select s.date, su.supplier, s.description, s.amount
+FROM spends s
+INNER JOIN expense_areas ea ON s.expense_area_id = ea.id
+INNER JOIN suppliers su ON s.supplier_id = su.id
+WHERE ea.expense_area = 'Better Hospital Food';

```

Comment:
while your query here is absolutely fine 
i just want to share my personal experience with SQL
when writing joins it better read queries if:
- `on` closure start on a new line and has left indentation
- joining condition starts with the joined field

 so having said that, i would format the query this way:
```sql
select 
    s.date, su.supplier, s.description, s.amount
FROM spends s
INNER JOIN expense_areas ea 
    ON ea.id = s.expense_area_id
INNER JOIN suppliers su 
    ON su.id = s.supplier_id
WHERE ea.expense_area = 'Better Hospital Food';
```
no difference in the way it works. just a matter of styling preferences.
some people also prefer having selected fields each on a separate line:
```sql
select 
    s.date, 
    su.supplier, 
    s.description, 
    s.amount
FROM spends s
```